Sources/iron/RenderPath.hx

```diff
@@ -689,14 +689,18 @@ class RenderPath {
 		if (height < 1) height = 1;
 		if (t.depth != null && t.depth > 1) { // 3D texture
 			// Image only
-			return Image.create3D(width, height, depth, t.format != null ? getTextureFormat(t.format) : TextureFormat.RGBA32);
+			var img = Image.create3D(width, height, depth,
+				t.format != null ? getTextureFormat(t.format) : TextureFormat.RGBA32);
+			if (t.mipmaps)
+				img.generateMipmaps(1000); // Allocate mipmaps
+			return img;
 		}
 		else { // 2D texture
 			if (t.is_image != null && t.is_image) { // Image
 				var img = Image.create(width, height,
 					t.format != null ? getTextureFormat(t.format) : TextureFormat.RGBA32);
 				if (t.mipmaps)
-					img.generateMipmaps(16); // Allocate mipmaps
+					img.generateMipmaps(1000); // Allocate mipmaps
```

Comment:
It is really confusing to review it like this, the value was originally 1000, then was changed to 16, and now it's back to 1000 in an unrelated commit. :)

Reply:
I know, actually I was trying to optimize the mipmaps before disabling it.
Let it be a thousand and anyways for voxels we don't use mipmaps.